sql/moz-fx-data-shared-prod/fenix_derived/google_ads_campaign_cost_breakdowns_v1/query.sql

```diff
@@ -0,0 +1,76 @@
+WITH campaigns AS (
+  SELECT DISTINCT
+    id,
+    name,
+    FORMAT("%s (%s)", name, CAST(id AS string)) AS fenix_compatible_campaign_name
+  FROM
+    `moz-fx-data-bq-fivetran`.google_ads.campaign_history
+),
+install_dou_metrics AS (
+  SELECT
+    fenix_marketing_metrics.adjust_campaign AS fenix_marketing_metrics_adjust_campaign,
+    fenix_marketing_metrics.submission_date AS date,
+    COALESCE(SUM(fenix_marketing_metrics.new_profiles), 0) AS new_profiles_sum,
+    COALESCE(SUM(fenix_marketing_metrics.dau), 0) AS dau_sum
+  FROM
+    `moz-fx-data-shared-prod.fenix.marketing_attributable_metrics` AS fenix_marketing_metrics
+  GROUP BY
+    fenix_marketing_metrics_adjust_campaign,
+    date
+)
+SELECT
+  campaigns.name AS campaign_name,
+  stats.date AS date,
+  -- Total spend per-campaign
+  stats.cost_micros AS campaign_spend_in_microcents,
+  -- Impressions and clicks over time for each campaign (clicks of our ads)
+  impressions AS ad_impressions,
+  install_dou_metrics.new_profiles_sum AS installs,
+  install_dou_metrics.dau_sum AS dous,
+  clicks AS ad_clicks,
+  -- Cost per-install for each campaign ($/new profiles)
+  CASE
+  WHEN
+    install_dou_metrics.new_profiles_sum = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / install_dou_metrics.new_profiles_sum
+  END
+  AS cost_per_install_microcents,
+  -- Cost per-DOU for each campaign ($/DOU)
+  CASE
+  WHEN
+    install_dou_metrics.dau_sum = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / install_dou_metrics.dau_sum
+  END
+  AS cost_per_dou_microcents,
+  -- Cost per-Ad Click for each campaign ($/Ad Clicks)
+  CASE
+  WHEN
+    clicks = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / clicks
+  END
+  AS cost_per_click_microcents
+FROM
+  `moz-fx-data-bq-fivetran`.google_ads.campaign_stats AS stats
```

Comment:
This table has more than 1 row per-campaign, we should aggregate it before joining here

Reply:
They're unique by date though, which we use to partition the metrics. Do we want to flatten that?

Reply:
Sorry - more than 1 row per-campaign & per-date. See e.g.
```
SELECT id, date, COUNT(*)
FROM `moz-fx-data-bq-fivetran`.google_ads.campaign_stats AS stats
GROUP BY 1, 2
ORDER BY 3 DESC
```

Reply:
Oy vey! Okie dokie, I'll group these in a CTE beforehend.